pandas/io/parsers/base_parser.py

```diff
@@ -207,6 +213,30 @@ def __init__(self, kwds):
 
         self.handles: Optional[IOHandles] = None
 
+        # Bad line handling
+        on_bad_lines = kwds.get("on_bad_lines", "error")
+        if on_bad_lines == "error":
+            self.on_bad_lines = self.BadLineHandleMethod.ERROR
+        elif on_bad_lines == "warn":
+            self.on_bad_lines = self.BadLineHandleMethod.WARN
+        elif on_bad_lines == "skip":
+            self.on_bad_lines = self.BadLineHandleMethod.SKIP
+        else:
+            raise ValueError(f"Argument {on_bad_lines} is invalid for on_bad_lines")
+        # Override on_bad_lines w/ deprecated args for backward compatibility
```

Comment:
shouldn't all of these cases show a deprecation warning? L227-238

Reply:
Like, I said before this is handled by _deprecated_defaults. Ideally _deprecated_defaults wouldn't exist, but as we all know, read_csv is very bloated in terms of API, so its nice to show the deprecation warnings from one place. If you are still unsure, this is tested in this PR here https://github.com/pandas-dev/pandas/pull/40413/files#diff-2f09043040b80b8e52cb8525b43b1653de0ce7d04e0999ca8e0dbe69b05b88faR758-R770